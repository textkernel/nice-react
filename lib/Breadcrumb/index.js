export { default as BreadcrumbGroup } from './BreadcrumbGroup';
export { default as BreadcrumbItem } from './BreadcrumbItem';