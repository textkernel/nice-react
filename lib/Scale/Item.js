import React from 'react';
import PropTypes from 'prop-types';

const Item = ({ label, value, className, name, checked, onSelect, tickLabel }) => {
    const handleSelect = e => onSelect(e, e.currentTarget.value);

    return (
      <li key={value}>
        <label>
          <input 
            type='radio'
            name={name}
            value={value}
            checked={checked}
            onChange={handleSelect}
          />
          <span />
          { 
            label &&
              <span className={`${className} ${tickLabel ? 'nice-scale-label' : ''}`}>
                {label}
              </span>
          }
        </label>
      </li>
    );
};

Item.propTypes = {
    label: PropTypes.string,
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
    className: PropTypes.string,
    name: PropTypes.string,
    tickLabel: PropTypes.bool,
    onSelect: PropTypes.func,
    checked: PropTypes.bool
};

Item.defaultProps = {
    label: '',
    className: '',
    name: '',
    tickLabel: true,
    onSelect: () => true,
    checked: false
};

export default Item;