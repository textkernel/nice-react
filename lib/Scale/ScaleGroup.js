import React from 'react';
import PropTypes from 'prop-types';

const ScaleGroup = ({ name, tickLabels, onChange, className, children }) => {
  const childrenWithProps = React.Children.map(children, child =>
    React.cloneElement(child, { 
      name,
      onSelect: onChange,
      tickLabel: tickLabels 
    })
  );

  return (
    <div className={`nice-scale ${className}`}>
      <ol className='nice-scale-selection'>
        {childrenWithProps}
      </ol>
    </div>
  );
};

ScaleGroup.propTypes = {
  name: PropTypes.string.isRequired,
  className: PropTypes.string,
  tickLabels: PropTypes.bool,
  onChange: PropTypes.func,
  children: PropTypes.node.isRequired
};

ScaleGroup.defaultProps = {
  className: '',
  tickLabels: true,
  onChange: () => true
};

export default ScaleGroup;