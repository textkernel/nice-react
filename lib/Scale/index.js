export { default as ScaleGroup } from './ScaleGroup';
export { default as ScaleItem } from './ScaleItem';