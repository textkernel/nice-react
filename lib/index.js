import '@textkernel/nice/dist/nice.min.css';

export { default as Panel } from './Panel';
export { Button, ButtonGroup } from './Buttons';
export { default as Progress } from './Progress';
export { default as Loading } from './Loading';
export { BreadcrumbGroup, BreadcrumbItem } from './Breadcrumb';