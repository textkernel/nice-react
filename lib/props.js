const contexts = ['brand', 'link', 'primary', 'accent', 'good', 'warning', 'bad'];
const sizes = ['small', 'medium', 'large'];
 
const props = {
  contexts,
  buttonContexts: contexts,
  buttonSizes: sizes 
};

module.exports = props;