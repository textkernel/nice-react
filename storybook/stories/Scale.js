import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { action } from '@storybook/addon-actions';

import { ScaleGroup, ScaleItem } from 'nice-react';

const stories = storiesOf('Scale', module);

stories.addDecorator((story, context) => withInfo('')(story)(context));

stories.add('with all labels', () => {
    return (
      <ScaleGroup name='projectid-radios' onChange={ action('radio-click') }> 
        <ScaleItem label='Nice to have' value={1} />
        <ScaleItem label='Good to have' value={5} />
        <ScaleItem label='Must have' value={10} />
      </ScaleGroup>
    );
});

stories.add('without one label', () => {
    return (
      <ScaleGroup name='projectid-radios' onChange={ action('radio-click') }> 
        <ScaleItem label='Nice to have' value={1} />
        <ScaleItem value={5} />
        <ScaleItem label='Must have' value={10} />
      </ScaleGroup>
    );
});

stories.add('without tick labels', () => {
    return (
      <ScaleGroup name='projectid-radios' tickLabels={false} onChange={ action('radio-click') }> 
        <ScaleItem label='Nice to have' value={1} />
        <ScaleItem value={5} />
        <ScaleItem label='Must have' value={10} />
      </ScaleGroup>
    );
});
  

stories.add('with custom classes over items', () => {
    const items = [
        { label: 'Nice to have', value: 1, className: 'text-bold' },
        { label: 'Good to have', value: 5, className: 'text-bold text-italic' },
        { label: 'Must have', value: 10, className: 'text-uppercase text-brand' }
    ];
    
    return (
      <ScaleGroup name='projectid-radios' tickLabels={false} onChange={ action('radio-click') }> 
        <ScaleItem label='Nice to have' value={1} className='text-bold'/>
        <ScaleItem value={5} className='text-bold text-italic' />
        <ScaleItem label='Must have' value={10} className='text-uppercase text-brand' />
      </ScaleGroup>
    );
});

stories.add('with custom classes over outermost element', () => {
    const items = [
        { label: 'Nice to have', value: 1 },
        { label: 'Good to have', value: 5 },
        { label: 'Must have', value: 10 }
    ];
    
    return (
      <ScaleGroup name='projectid-radios' className='nice-bg-info' onChange={ action('radio-click') }> 
        <ScaleItem label='Nice to have' value={1} />
        <ScaleItem value={5} />
        <ScaleItem label='Must have' value={10} />
      </ScaleGroup>
    );
});