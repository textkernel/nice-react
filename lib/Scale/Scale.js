import React from 'react';
import PropTypes from 'prop-types';
import Item from './Item';

const Scale = ({ name, tickLabels, onChange, className, children, value }) => {
    const childrenWithProps = React.Children.map(children, child =>
      React.cloneElement(child, { 
          name,
          checked: child.props.value === value,
          onSelect: onChange,
          tickLabel: tickLabels 
      })
    );

    return (
      <div className={`nice-scale ${className}`}>
        <ol className='nice-scale-selection'>
          {childrenWithProps}
        </ol>
      </div>
    );
};

Scale.Item = Item;

Scale.propTypes = {
    name: PropTypes.string.isRequired,
    className: PropTypes.string,
    tickLabels: PropTypes.bool,
    onChange: PropTypes.func,
    children: PropTypes.node.isRequired,
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
};

Scale.defaultProps = {
    className: '',
    tickLabels: true,
    onChange: () => true,
    value: ''
};

export default Scale;