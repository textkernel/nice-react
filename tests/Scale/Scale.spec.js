import React from 'react';
import { ScaleGroup, ScaleItem } from '../../lib/Scale';

describe('<ScaleGroup /> <ScaleItem />', () => {
    it('should render with all labels', () => {
        const wrapper = mount(
          <ScaleGroup name='my-radio-name'> 
            <ScaleItem label='Nice to have' value={1} />
            <ScaleItem label='Good to have' value={5} />
            <ScaleItem label='Must have' value={10} />
          </ScaleGroup>
      );
        
        expect(wrapper).toMatchSnapshot();
    });
});

describe('<ScaleGroup /> <ScaleItem />', () => {
    it('should render without one label', () => {
        const wrapper = mount(
          <ScaleGroup name='my-radio-name'> 
            <ScaleItem label='Nice to have' value={1} />
            <ScaleItem value={5} />
            <ScaleItem label='Must have' value={10} />
          </ScaleGroup>
        );
    
        expect(wrapper).toMatchSnapshot();
    });
});

describe('<ScaleGroup /> <ScaleItem />', () => {
    it('should render as controlled component', () => {
        const wrapper = mount(
          <ScaleGroup name='my-radio-name' value={10}> 
            <ScaleItem label='Nice to have' value={1} />
            <ScaleItem value={5} />
            <ScaleItem label='Must have' value={10} />
          </ScaleGroup>
        );
    
        expect(wrapper).toMatchSnapshot();
    });
});

describe('<ScaleGroup /> <ScaleItem />', () => {
    it('should render without tick labels', () => {
        const wrapper = mount(
          <ScaleGroup tickLabels={false} name='my-radio-name'> 
            <ScaleItem label='Nice to have' value={1} />
            <ScaleItem value={5} />
            <ScaleItem label='Must have' value={10} />
          </ScaleGroup>
        );
        
        expect(wrapper).toMatchSnapshot();
    });
});

describe('<ScaleGroup /> <ScaleItem />', () => {
    it('should render with custom classes over items', () => {
        const wrapper = mount(
          <ScaleGroup name='projectid-radios'>
            <ScaleItem label='Nice to have' value={1} className='text-bold' />
            <ScaleItem value={5} className='text-bold text-italic' />
            <ScaleItem label='Must have' value={10} className='text-uppercase text-brand' />
          </ScaleGroup>
        );
        
        expect(wrapper).toMatchSnapshot();
    });
});

describe('<ScaleGroup /> <ScaleItem />', () => {
    it('should render with custom classes over outermost element', () => {
        const wrapper = mount(
          <ScaleGroup name='projectid-radios' className='nice-bg-info'>
            <ScaleItem label='Nice to have' value={1} />
            <ScaleItem value={5} />
            <ScaleItem label='Must have' value={10} />
          </ScaleGroup>
        );
        
        expect(wrapper).toMatchSnapshot();
    });
});

describe('<ScaleGroup /> <ScaleItem />', () => {
    it('should handle the onChange event', () => {
        const onChangeHandler = jest.fn();

        const wrapper = mount(
          <ScaleGroup name='projectid-radios' onChange={onChangeHandler}>
            <ScaleItem label='Nice to have' value={1} />
            <ScaleItem value={5} />
            <ScaleItem label='Must have' value={10} />
          </ScaleGroup>
        );
        
        wrapper.find('input').at(0).simulate('change');
        expect(onChangeHandler.mock.calls[0][1]).toBe('1');

        wrapper.find('input').at(1).simulate('change');
        expect(onChangeHandler.mock.calls[1][1]).toBe('5');

        wrapper.find('input').at(2).simulate('change');
        expect(onChangeHandler.mock.calls[2][1]).toBe('10');
    });
});