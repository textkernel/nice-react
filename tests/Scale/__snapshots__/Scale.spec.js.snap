// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScaleGroup /> <ScaleItem /> should render as controlled component 1`] = `
<ScaleGroup
  className=""
  name="my-radio-name"
  onChange={[Function]}
  tickLabels={true}
  value={10}
>
  <div
    className="nice-scale "
  >
    <ol
      className="nice-scale-selection"
    >
      <ScaleItem
        checked={false}
        className=""
        key=".0"
        label="Nice to have"
        name="my-radio-name"
        onSelect={[Function]}
        tickLabel={true}
        value={1}
      >
        <li
          key="1"
        >
          <label>
            <input
              checked={false}
              name="my-radio-name"
              onChange={[Function]}
              type="radio"
              value={1}
            />
            <span />
            <span
              className=" nice-scale-label"
            >
              Nice to have
            </span>
          </label>
        </li>
      </ScaleItem>
      <ScaleItem
        checked={false}
        className=""
        key=".1"
        label=""
        name="my-radio-name"
        onSelect={[Function]}
        tickLabel={true}
        value={5}
      >
        <li
          key="5"
        >
          <label>
            <input
              checked={false}
              name="my-radio-name"
              onChange={[Function]}
              type="radio"
              value={5}
            />
            <span />
          </label>
        </li>
      </ScaleItem>
      <ScaleItem
        checked={true}
        className=""
        key=".2"
        label="Must have"
        name="my-radio-name"
        onSelect={[Function]}
        tickLabel={true}
        value={10}
      >
        <li
          key="10"
        >
          <label>
            <input
              checked={true}
              name="my-radio-name"
              onChange={[Function]}
              type="radio"
              value={10}
            />
            <span />
            <span
              className=" nice-scale-label"
            >
              Must have
            </span>
          </label>
        </li>
      </ScaleItem>
    </ol>
  </div>
</ScaleGroup>
`;

exports[`<ScaleGroup /> <ScaleItem /> should render with all labels 1`] = `
<ScaleGroup
  className=""
  name="my-radio-name"
  onChange={[Function]}
  tickLabels={true}
  value=""
>
  <div
    className="nice-scale "
  >
    <ol
      className="nice-scale-selection"
    >
      <ScaleItem
        checked={false}
        className=""
        key=".0"
        label="Nice to have"
        name="my-radio-name"
        onSelect={[Function]}
        tickLabel={true}
        value={1}
      >
        <li
          key="1"
        >
          <label>
            <input
              checked={false}
              name="my-radio-name"
              onChange={[Function]}
              type="radio"
              value={1}
            />
            <span />
            <span
              className=" nice-scale-label"
            >
              Nice to have
            </span>
          </label>
        </li>
      </ScaleItem>
      <ScaleItem
        checked={false}
        className=""
        key=".1"
        label="Good to have"
        name="my-radio-name"
        onSelect={[Function]}
        tickLabel={true}
        value={5}
      >
        <li
          key="5"
        >
          <label>
            <input
              checked={false}
              name="my-radio-name"
              onChange={[Function]}
              type="radio"
              value={5}
            />
            <span />
            <span
              className=" nice-scale-label"
            >
              Good to have
            </span>
          </label>
        </li>
      </ScaleItem>
      <ScaleItem
        checked={false}
        className=""
        key=".2"
        label="Must have"
        name="my-radio-name"
        onSelect={[Function]}
        tickLabel={true}
        value={10}
      >
        <li
          key="10"
        >
          <label>
            <input
              checked={false}
              name="my-radio-name"
              onChange={[Function]}
              type="radio"
              value={10}
            />
            <span />
            <span
              className=" nice-scale-label"
            >
              Must have
            </span>
          </label>
        </li>
      </ScaleItem>
    </ol>
  </div>
</ScaleGroup>
`;

exports[`<ScaleGroup /> <ScaleItem /> should render with custom classes over items 1`] = `
<ScaleGroup
  className=""
  name="projectid-radios"
  onChange={[Function]}
  tickLabels={true}
  value=""
>
  <div
    className="nice-scale "
  >
    <ol
      className="nice-scale-selection"
    >
      <ScaleItem
        checked={false}
        className="text-bold"
        key=".0"
        label="Nice to have"
        name="projectid-radios"
        onSelect={[Function]}
        tickLabel={true}
        value={1}
      >
        <li
          key="1"
        >
          <label>
            <input
              checked={false}
              name="projectid-radios"
              onChange={[Function]}
              type="radio"
              value={1}
            />
            <span />
            <span
              className="text-bold nice-scale-label"
            >
              Nice to have
            </span>
          </label>
        </li>
      </ScaleItem>
      <ScaleItem
        checked={false}
        className="text-bold text-italic"
        key=".1"
        label=""
        name="projectid-radios"
        onSelect={[Function]}
        tickLabel={true}
        value={5}
      >
        <li
          key="5"
        >
          <label>
            <input
              checked={false}
              name="projectid-radios"
              onChange={[Function]}
              type="radio"
              value={5}
            />
            <span />
          </label>
        </li>
      </ScaleItem>
      <ScaleItem
        checked={false}
        className="text-uppercase text-brand"
        key=".2"
        label="Must have"
        name="projectid-radios"
        onSelect={[Function]}
        tickLabel={true}
        value={10}
      >
        <li
          key="10"
        >
          <label>
            <input
              checked={false}
              name="projectid-radios"
              onChange={[Function]}
              type="radio"
              value={10}
            />
            <span />
            <span
              className="text-uppercase text-brand nice-scale-label"
            >
              Must have
            </span>
          </label>
        </li>
      </ScaleItem>
    </ol>
  </div>
</ScaleGroup>
`;

exports[`<ScaleGroup /> <ScaleItem /> should render with custom classes over outermost element 1`] = `
<ScaleGroup
  className="nice-bg-info"
  name="projectid-radios"
  onChange={[Function]}
  tickLabels={true}
  value=""
>
  <div
    className="nice-scale nice-bg-info"
  >
    <ol
      className="nice-scale-selection"
    >
      <ScaleItem
        checked={false}
        className=""
        key=".0"
        label="Nice to have"
        name="projectid-radios"
        onSelect={[Function]}
        tickLabel={true}
        value={1}
      >
        <li
          key="1"
        >
          <label>
            <input
              checked={false}
              name="projectid-radios"
              onChange={[Function]}
              type="radio"
              value={1}
            />
            <span />
            <span
              className=" nice-scale-label"
            >
              Nice to have
            </span>
          </label>
        </li>
      </ScaleItem>
      <ScaleItem
        checked={false}
        className=""
        key=".1"
        label=""
        name="projectid-radios"
        onSelect={[Function]}
        tickLabel={true}
        value={5}
      >
        <li
          key="5"
        >
          <label>
            <input
              checked={false}
              name="projectid-radios"
              onChange={[Function]}
              type="radio"
              value={5}
            />
            <span />
          </label>
        </li>
      </ScaleItem>
      <ScaleItem
        checked={false}
        className=""
        key=".2"
        label="Must have"
        name="projectid-radios"
        onSelect={[Function]}
        tickLabel={true}
        value={10}
      >
        <li
          key="10"
        >
          <label>
            <input
              checked={false}
              name="projectid-radios"
              onChange={[Function]}
              type="radio"
              value={10}
            />
            <span />
            <span
              className=" nice-scale-label"
            >
              Must have
            </span>
          </label>
        </li>
      </ScaleItem>
    </ol>
  </div>
</ScaleGroup>
`;

exports[`<ScaleGroup /> <ScaleItem /> should render without one label 1`] = `
<ScaleGroup
  className=""
  name="my-radio-name"
  onChange={[Function]}
  tickLabels={true}
  value=""
>
  <div
    className="nice-scale "
  >
    <ol
      className="nice-scale-selection"
    >
      <ScaleItem
        checked={false}
        className=""
        key=".0"
        label="Nice to have"
        name="my-radio-name"
        onSelect={[Function]}
        tickLabel={true}
        value={1}
      >
        <li
          key="1"
        >
          <label>
            <input
              checked={false}
              name="my-radio-name"
              onChange={[Function]}
              type="radio"
              value={1}
            />
            <span />
            <span
              className=" nice-scale-label"
            >
              Nice to have
            </span>
          </label>
        </li>
      </ScaleItem>
      <ScaleItem
        checked={false}
        className=""
        key=".1"
        label=""
        name="my-radio-name"
        onSelect={[Function]}
        tickLabel={true}
        value={5}
      >
        <li
          key="5"
        >
          <label>
            <input
              checked={false}
              name="my-radio-name"
              onChange={[Function]}
              type="radio"
              value={5}
            />
            <span />
          </label>
        </li>
      </ScaleItem>
      <ScaleItem
        checked={false}
        className=""
        key=".2"
        label="Must have"
        name="my-radio-name"
        onSelect={[Function]}
        tickLabel={true}
        value={10}
      >
        <li
          key="10"
        >
          <label>
            <input
              checked={false}
              name="my-radio-name"
              onChange={[Function]}
              type="radio"
              value={10}
            />
            <span />
            <span
              className=" nice-scale-label"
            >
              Must have
            </span>
          </label>
        </li>
      </ScaleItem>
    </ol>
  </div>
</ScaleGroup>
`;

exports[`<ScaleGroup /> <ScaleItem /> should render without tick labels 1`] = `
<ScaleGroup
  className=""
  name="my-radio-name"
  onChange={[Function]}
  tickLabels={false}
  value=""
>
  <div
    className="nice-scale "
  >
    <ol
      className="nice-scale-selection"
    >
      <ScaleItem
        checked={false}
        className=""
        key=".0"
        label="Nice to have"
        name="my-radio-name"
        onSelect={[Function]}
        tickLabel={false}
        value={1}
      >
        <li
          key="1"
        >
          <label>
            <input
              checked={false}
              name="my-radio-name"
              onChange={[Function]}
              type="radio"
              value={1}
            />
            <span />
            <span
              className=" "
            >
              Nice to have
            </span>
          </label>
        </li>
      </ScaleItem>
      <ScaleItem
        checked={false}
        className=""
        key=".1"
        label=""
        name="my-radio-name"
        onSelect={[Function]}
        tickLabel={false}
        value={5}
      >
        <li
          key="5"
        >
          <label>
            <input
              checked={false}
              name="my-radio-name"
              onChange={[Function]}
              type="radio"
              value={5}
            />
            <span />
          </label>
        </li>
      </ScaleItem>
      <ScaleItem
        checked={false}
        className=""
        key=".2"
        label="Must have"
        name="my-radio-name"
        onSelect={[Function]}
        tickLabel={false}
        value={10}
      >
        <li
          key="10"
        >
          <label>
            <input
              checked={false}
              name="my-radio-name"
              onChange={[Function]}
              type="radio"
              value={10}
            />
            <span />
            <span
              className=" "
            >
              Must have
            </span>
          </label>
        </li>
      </ScaleItem>
    </ol>
  </div>
</ScaleGroup>
`;
